src/colors.rs

```diff
@@ -23,6 +23,14 @@ use termcolor::BufferWriter;
 use termcolor::ColorChoice;
 
 static USE_COLOR: Lazy<AtomicBool> = Lazy::new(|| {
+  #[cfg(non_tty)]
+  {
+    // Don't use color, by default, when not in a TTY, because text becomes
+    // illegible when redirected to a file or another program.
+    //
+    // Instead the user can opt-in via `set_use_color`.
+    AtomicBool::new(false)
+  }
```

Comment:
I think this will be ignored in the end because one of the expressions below will take precedence?

Reply:
What do you suggest we do? Could we use `is_stdout_tty()` in `use_color()`?

Reply:
Oh, actually is this code used with `console.log`? If so, `console.log` should only take if stdout is a tty into account and `console.error` should only take into account if stderr is a tty.

Reply:
Yeah, AFAIK, this code is used to color any text in the runtime. Cool. I was thinking we need to check for stderr and stdout being a TTY, too.

Reply:
Looks like this is already fixed in Deno: https://github.com/denoland/deno/issues/23668#issuecomment-2166697214